atom/browser/net/atom_network_delegate.cc

```diff
@@ -75,9 +75,15 @@ void ToDictionary(base::DictionaryValue* details, net::URLRequest* request) {
   details->SetInteger("id", request->identifier());
   details->SetDouble("timestamp", base::Time::Now().ToDoubleT() * 1000);
   auto info = content::ResourceRequestInfo::ForRequest(request);
-  details->SetString("resourceType",
-                     info ? ResourceTypeToString(info->GetResourceType())
-                          : "other");
+  if (info) {
+    int64_t process_id = info->GetChildID();
+    int64_t routing_id = info->GetRouteID();
+    details->SetDouble("webContentsGetId", (process_id << 32) + routing_id);
```

Comment:
This should probably be just `webContentsId`

And assuming my understanding of this bit shifting is correct that ID should just work with https://electron.atom.io/docs/api/web-contents/#webcontentsfromidid 👍

Reply:
- `webContentsId` is actually the `ID` in a weak map, as shown [here](https://github.com/electron/electron/blob/bc39964ece14a428c4efdb23084f78737f25f681/atom/browser/api/trackable_object.h#L28).
- `webContentsGetId` is, in the other hand, a 64-bit integer composed of both the process id(high 32), and the RenderViewHost routing id(low 32).

Thus, the argument passed to `webContents.fromId(id)` should be `webContentsId`, not `webContentsGetId`. Besides, there's no api to get back the corresponding webContents from `webContentsGetId` afaik 😭 .

Reply:
I think you can use the process ID and routing ID to get the WebContents with:

```c++
WebContents::FromRenderFrameHost(RenderFrameHost::FromID(process_id, routing_id));
```